perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -295,21 +297,94 @@ public Group call(ApiCaller ac, Deserializer parms) throws PerunException {
 		}
 	},
 
+
+	/*#
+	 * Adds members to a group. If already a member of the group, the member will be skipped.
+	 * Non-empty list of members expected, if empty, no member will be added.
+	 *
+	 * @param groups List<Group> List of groups that the member should be added to <code>id</code>
+	 * @param member int Member <code>id</code>
+	 */
+	addMembers {
+		@Override
+		public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
+			ac.stateChangingCheck();
+
+			List<Integer> memberInts = parms.readList("members", Integer.class);
+			if (memberInts == null) {
+				throw new RpcException(RpcException.Type.MISSING_VALUE, "Non-empty list of members not sent.");
+			}
+			Group group = ac.getGroupById(parms.readInt("group"));
+			List<Member> members = new ArrayList<>();
+			for (Integer memberInt : memberInts) {
+				Member member = ac.getMemberById(memberInt);
+				if (!ac.getGroupsManager().isGroupMember(ac.getSession(), group, member)) {
+					members.add(member);
+				} else {
+					Member groupMember = ac.getGroupsManager().getGroupMemberById(ac.getSession(), group, member.getId());
+					if (MembershipType.INDIRECT.equals(groupMember.getMembershipType())) {
+						members.add(member);
+					}
+				}
+			}
+			ac.getGroupsManager().addMembers(ac.getSession(),
+				group,
+				members);
+
+			return null;
+		}
+	},
+
+
+
 	/*#
 	 * Adds a member to a group.
 	 *
 	 * @param group int Group <code>id</code>
 	 * @param member int Member <code>id</code>
 	 */
+	/*#
+	 * Adds a member to a group. If already a member of a group, the group will be skipped.
+	 * Non-empty list of groups expected, if empty, no member will be added.
+	 *
+	 * @param groups List<Group> List of groups that the member should be added to <code>id</code>
+	 * @param member int Member <code>id</code>
+	 */
 	addMember {
 
 		@Override
 		public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 			ac.stateChangingCheck();
 
-			ac.getGroupsManager().addMember(ac.getSession(),
+			if (parms.contains("group")) {
+				ac.getGroupsManager().addMember(ac.getSession(),
 					ac.getGroupById(parms.readInt("group")),
 					ac.getMemberById(parms.readInt("member")));
+			} else if (parms.contains("groups")) {
+				List<Integer> groupsInts = parms.readList("groups", Integer.class);
+				if (groupsInts == null) {
+					throw new RpcException(RpcException.Type.MISSING_VALUE, "Non-empty list of groups not sent.");
+				}
+				List<Group> groups = new ArrayList<>();
+				Member member = ac.getMemberById(parms.readInt("member"));
+				for (Integer groupInt : groupsInts) {
+					Group group = ac.getGroupById(groupInt);
+					if (!ac.getGroupsManager().isGroupMember(ac.getSession(), group, member)) {
```

Comment:
As in previous method, we could save up select by using ``isDirectGroupMember``.